lib/reek/spec.rb

```diff
@@ -85,8 +85,10 @@ module Spec
     #
     #   expect(src).to reek_of(Reek::Smells::DuplicateMethodCall, name: '@other.thing')
     #
-    def reek_of(smell_category, smell_details = {})
-      ShouldReekOf.new(smell_category, smell_details)
+    def reek_of(smell_category,
+                smell_details = {},
+                configuration = Configuration::AppConfiguration.new)
```

Comment:
API change, but none breaking. @mvz suggested to go with something like:

```Ruby
expect('class Foo; end').to reek_of(:DataClump).with_config(...)
```

But I couldn't find anything obvious regarding this topic. Since it's not breaking, we can go with this solution and then maybe improve this later.